pkg/operator/ceph/cluster/dependents.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util"
+	"github.com/rook/rook/pkg/util/dependents"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	// must use plural kinds
+	cephClusterDependentPluralKinds []string = []string{
+		"CephBlockPools",
+		"CephRBDMirrors",
+		"CephFilesystems",
+		"CephFilesystemMirrors",
+		"CephObjectStores",
+		"CephObjectStoreUsers",
+		"CephObjectZones",
+		"CephObjectZoneGroups",
+		"CephObjectRealms",
+		"CephNFSes",
+		"CephClients",
+	}
+)
+
+// CephClusterDependents returns a DependentList of dependents of a CephCluster in the namespace.
+func CephClusterDependents(c *clusterd.Context, namespace string) (*dependents.DependentList, error) {
+	ctx := context.TODO()
+
+	dependents := dependents.NewDependentList()
+	errs := []error{}
+
+	for _, pluralKind := range cephClusterDependentPluralKinds {
+		resource := pluralKindToResource(pluralKind)
+		gvr := cephv1.SchemeGroupVersion.WithResource(resource)
+		list, err := c.DynamicClientset.Resource(gvr).Namespace(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
Also, I wonder how we could prevent the deletion of the CephCluster if it is consumed externally by another cluster. We could fetch all CephCluster from all namespaces and look at their Spec to see if it's external or not. We might not need this for this PR but I'd love to have something like this in a follow-up PR.

So `CephCluster` could be part of `cephClusterDependentPluralKinds` I believe.

Reply:
Is there a production scenario where the external cluster CR would be in the same K8s cluster as the main cluster CR? I thought it would always be in a different cluster.

Reply:
Not that I have seen. I guess I've been thinking too much and also because that's how we test in the CI 😄 . Let's forget about this.